src/lib.rs

```diff
@@ -0,0 +1,310 @@
+
+        let rustdoc_paths = match &self.current {
+            Current::RustDoc(rustdoc_path) => {
+                let name = "<unknown>";
+                let version = None;
+                vec![(
+                    name.to_owned(),
+                    loader.load_rustdoc(&mut config, &rustdoc_cmd, name, version)?,
+                    rustdoc_path.to_owned(),
+                )]
+            }
+            Current::CurrentDir | Current::Manifest(_) => {
+                let metadata = self.manifest_metadata()?;
+                let selected = self.scope.selected_packages(&metadata);
+                let mut rustdoc_paths = Vec::with_capacity(selected.len());
+                for selected in selected {
+                    let manifest_path = selected.manifest_path.as_std_path();
+                    let crate_name = &selected.name;
+                    let version = &selected.version;
+
+                    let is_implied = self.scope.selection == ScopeSelection::Workspace;
+                    if is_implied && selected.publish == Some(vec![]) {
+                        config.verbose(|config| {
+                            config.shell_status(
+                                "Skipping",
+                                format_args!("{crate_name} v{version} (current)"),
+                            )
+                        })?;
+                        continue;
+                    }
+
+                    config.shell_status(
+                        "Parsing",
+                        format_args!("{crate_name} v{version} (current)"),
+                    )?;
+                    let rustdoc_path = rustdoc_cmd.dump(manifest_path, None, true)?;
+                    let baseline_path = loader.load_rustdoc(
+                        &mut config,
+                        &rustdoc_cmd,
+                        crate_name,
+                        Some(version),
+                    )?;
+                    rustdoc_paths.push((crate_name.clone(), baseline_path, rustdoc_path));
+                }
+                rustdoc_paths
+            }
+        };
+        let mut success = true;
+        for (crate_name, baseline_path, current_path) in rustdoc_paths {
+            let baseline_crate = load_rustdoc(&baseline_path)?;
+            let current_crate = load_rustdoc(&current_path)?;
+
+            if !run_check_release(&mut config, &crate_name, current_crate, baseline_crate)? {
+                success = false;
+            }
+        }
+
+        Ok(Report { success })
+    }
+
+    fn registry_baseline(
+        &self,
+        config: &mut GlobalConfig,
+    ) -> Result<baseline::RegistryBaseline, anyhow::Error> {
+        let metadata = self.manifest_metadata_no_deps()?;
+        let target = metadata.target_directory.as_std_path().join(util::SCOPE);
+        let registry = baseline::RegistryBaseline::new(&target, config)?;
+        Ok(registry)
+    }
+}
+
+pub struct Report {
+    success: bool,
+}
```

Comment:
the report should contain the failing checks.
For now, the library prints to stdout and just reports if it fails or succeeds.